README.md

```diff
@@ -1,186 +1,87 @@
+## Why
 
-    ```python
-    # Arrange
-    pipeline: PipelineResource = test_framework.repository.get_pipeline_by_name("batch_job")
-
-    # Runs the pipeline with the provided parameters
-    activities = test_framework.evaluate_pipeline(pipeline, [
-        RunParameter(RunParameterType.Pipeline, "JobId", "123"),
-        RunParameter(RunParameterType.Pipeline, "ContainerName", "test-container"),
-        RunParameter(RunParameterType.Global, "BaseUrl", "https://example.com"),
-    ])
-
-    set_variable_activity: Activity = next(activities)
-    assert set_variable_activity is not None
-    assert "Set JobName" == set_variable_activity.name
-    assert "JobName" == activity.type_properties["variableName"]
-    assert "Job-123" == activity.type_properties["value"].value
-
-    get_version_activity = next(activities)
-    assert get_version_activity is not None
-    assert "Get version" == get_version_activity.name
-    assert "https://example.com/version" == get_version_activity.type_properties["url"].value
-    assert "GET" == get_version_activity.type_properties["method"]
-    get_version_activity.set_result(DependencyCondition.Succeeded,{"Version": "version1"})
-
-    create_batch_activity = next(activities)
-    assert create_batch_activity is not None
-    assert "Trigger Azure Batch Job" == create_batch_activity.name
-    assert "https://example.com/jobs" == create_batch_activity.type_properties["url"].value
-    assert "POST" == create_batch_activity.type_properties["method"]
-    body = create_batch_activity.type_properties["body"].get_json_value()
-    assert "123" == body["JobId"]
-    assert "Job-123" == body["JobName"]
-    assert "version1" == body["Version"]
-
-    with pytest.raises(StopIteration):
-        next(activities)
-    ```
-
-> See the [Examples](/examples) folder for more samples
-
-## Registering missing expression functions
-
-As the framework is interpreting expressions containing functions, these functions are implemented in the framework,
-but there may be bugs in some of them. You can override their implementation through:
+Data Factory does not support unit testing, nor testing your pipelines locally. Having integration and e2e tests running on an actual Data Factory instance is great, but having unit tests on top of them provides additional means of quick iteration, validation and regression testing. Unit testing with the _Data Factory Testing Framework_ has the following benefits:
 
-```python
-   FunctionsRepository.register("concat", lambda arguments: "".join(arguments))
-   FunctionsRepository.register("trim", lambda text, trim_argument: text.strip(trim_argument[0]))
-```
+* Runs locally with immediate feedback
+* Easier to cover a lot of different scenarios and edge cases
+* Regression testing
+
+## Concepts
+
+The following pages go deeper into different topics and concepts of the framework to help in getting you started.
+
+### Basic
+
+1. [Repository setup](docs/basic/repository_setup.md)
```

Comment:
Give we using relative links, maybe we need to fix something as this also ends up on pip - check a few examples here. These links don't work there. https://pypi.org/project/data-factory-testing-framework/

Reply:
Assumption we need to add these to the package :)

Reply:
Yeah, not sure how to approach here. Perhaps we can use a dedicated README.md for the package and link back to the repo?

Reply:
Maybe just adding the `doc` folder here is sufficient: https://github.com/microsoft/data-factory-testing-framework/blob/a44468fd441c9cdc7134b87bada2be2262818324/pyproject.toml#L8

On the other hand, maybe a `readthedocs` page the more practical approach if documentatin goes beyond a fewmd files. What do you think?

Reply:
I have tested to include "docs", but that didn't help. I indeed would like a readthedocs or similar page, but let's address that in a future PR.

Reply:
Let's track this as an issue, so it doesn't get lost in that case?

Reply:
https://github.com/orgs/microsoft/projects/961/views/1?pane=issue&itemId=52790377